test/runtests.jl

```diff
@@ -12,28 +19,71 @@ struct Foo
 end
 
 # test basic match
-x = nothing
-@test (@match Foo(x,2) = Foo(1,2)) == Foo(1,2)
-@test x == 1
+begin
+    local x = nothing
+    @test (@match Foo(x,2) = Foo(1,2)) == Foo(1,2)
+    @test x == 1
+end
```

Comment:
nit: maybe `let x = nothing` is simpler than `begin local x = nothing`?  But either way is fine i think.

Reply:
Another option would be to use `@testset`s to delineate logical groupings of tests, since that also introduces scope.

But we can also make that "cleanup" type change in a separate PR later.

Reply:
(Also, same for the others, below)